Sources/MarathonCore/Export.swift

```diff
@@ -0,0 +1,175 @@
+/**
+ *  Marathon
+ *  Copyright (c) John Sundell 2017
+ *  Licensed under the MIT license. See LICENSE file.
+ */
+
+import Foundation
+import Files
+
+// MARK: - Error
+
+public enum ExportError {
+    case missingPath
+    case noMarathonScriptFound
+    case failedToExportScript(String)
+}
+
+extension ExportError: PrintableError {
+    public var message: String {
+        switch self {
+        case .missingPath:
+            return "No script name/path given"
+        case .noMarathonScriptFound:
+            return "No marathon scripts found"
+        case .failedToExportScript(let name):
+            return "Failed to export script \(name)"
+        }
+    }
+
+    public var hints: [String] {
+        switch self {
+        case .missingPath:
+            return ["Pass the name/path of a script file to export (for example 'marathon export myScript')"]
+        case .noMarathonScriptFound:
+            return ["Marathon can only export scripts managed by marathon"]
+        case .failedToExportScript:
+            return ["Remove directory or provide alternate export path"]
+        }
+    }
+}
+
+// MARK: - Task
+
+internal final class ExportTask: Task, Executable {
+    private typealias Error = ExportError
+
+    func execute() throws {
+
+        // Required argument given?
+        guard let scriptNameOrPath = arguments.first?.asScriptPath() else {
+            throw Error.missingPath
+        }
+
+        // Script Name or Full Script Path given?
+        let isScriptName = scriptNameOrPath.components(separatedBy: "/").count == 1
```

Comment:
@JohnSundell You suggested using `scriptManager.script(at:)` instead of this. Assuming the following:

1. command is `marathon export myScript`
2. `myScript.swift` exists in current directory
3. `myScript.swift` is not a script managed by marathon

The following will happen:

1. marathon will add `myScript.swift` to its `List` of managed scripts, i.e., `marathon list`
2. marathon will add `myScript.swift` to its cache directory

Is this what we want? We're essentially allowing marathon to be used as just a package builder. `export` will work under these circumstances as it doesn't technically need to be a marathon-managed script to be exported.

Please advise.